cachecontrol/controller.py

```diff
@@ -164,9 +164,14 @@ def cached_request(self, request):
             return False
 
         now = time.time()
-        date = calendar.timegm(
-            parsedate_tz(headers['date'])
-        )
+
+        if 'date' in headers:
+            date = calendar.timegm(
+                parsedate_tz(headers['date'])
+            )
+        else:
+            date = 0
+
```

Comment:
The `if not headers or 'data' no in headers` on line 157 should ensure you have a `headers['date']` to parse. If you need a default date, then I'd move it before line 157. Does that make sense?